pkg/util/version/stream.go

```diff
@@ -14,33 +14,36 @@ type Stream struct {
 }
 
 // GetUpgradeStream determines if a valid upgrade path is available, and if so, returns the corresponding stream.
-func GetUpgradeStream(v *Version) (*Stream, error) {
-	// ARO version matches OCP version - X.Y.Z.
+func GetUpgradeStream(v *Version, upgradeY bool) (*Stream, error) {
+	// ARO version matches OCP version - X.Y.Z
 	// We know we can have only single version configured per Y release. These
-	// version should be edge points for Y+1 upgrades.
+	// versions should be edge points for Y+1 upgrades.
 
-	// We check first with which configured Y stream we are dealing with
+	// We first check which configured Y stream we are dealing with
 	for _, upgradeCandidate := range Streams {
-		if upgradeCandidate.Version.V[0] == v.V[0] &&
-			upgradeCandidate.Version.V[1] == v.V[1] {
+		if upgradeCandidate.Version.V[0] == v.V[0] && // Major (X) version is the same
+			upgradeCandidate.Version.V[1] == v.V[1] { // Minor (Y) version is the same
 
-			// we DO NOT upgrade if CVO version is already higher
+			// We DO NOT upgrade if CVO (cluster version operator) version is already higher
 			if upgradeCandidate.Version != nil && upgradeCandidate.Version.Lt(v) {
```

Comment:
`upgradeCandidate.Version != nil && ` is superfluous - we have already dereferenced it in the if statement above